### PR TITLE
fix Python wrappers warnings

### PR DESCRIPTION
**Pull request by davydden**

### CODE REVIEW

File: contrib/python-bindings/source/cell_accessor_wrapper.cc [hunk 1]
```diff
@@ -400,7 +400,7 @@ namespace python
 
   void CellAccessorWrapper::set_material_id(const int material_id)
   {
-    AssertThrow(material_id < numbers::invalid_material_id,
+    AssertThrow(static_cast<const types::material_id>(material_id) < numbers::invalid_material_id,
```
Comment (drwells) [inline contrib/python-bindings/source/cell_accessor_wrapper.cc]: Why do we store the material ID as an `int`?

Comment (davydden) [reply contrib/python-bindings/source/cell_accessor_wrapper.cc]: no idea, maybe @Rombur recalls

**Comment by drwells** [comment]
I confirmed that this fixes the warnings (and that we do not use `updated` anywhere).

File: contrib/python-bindings/source/cell_accessor_wrapper.cc (hunk 1 above)
Comment (Rombur) [reply contrib/python-bindings/source/cell_accessor_wrapper.cc]: I think we could store it as an `unsigned int` (or a `types::material_id`) but the range won't be the same as it is in C++. There is at least one mesh generator that uses a material ID of -1 for empty cells but boost throws when you try to extract a negative `unsigned int`, so I was forced to extract an `int`. That is why the material ID is an `int`.

**Comment by drwells** [comment]
Thats reasonable.

**Pull request merged**